File: bindings/scala/bench/src/main/scala/dev/everanium/itb/bench/BenchMessage.scala
// encryptMessage throughput vs plaintext size (Single Message
// profile) at 1 MiB / 16 MiB / 64 MiB.

package dev.everanium.itb.bench

import scala.util.Using

import dev.everanium.itb.Pipeline

object BenchMessage:

  def run(): Unit =
    val profile = BenchUtil.profileName("singlemsg-triple-nomac-v1")
    Using.resource(
      Pipeline.init(profile, BenchUtil.buildOpts).fold(e => throw e, identity)
    ) { pipe =>
      BenchUtil.header()
      for size <- BenchUtil.Sizes do
        val plain = BenchUtil.payload(size)
        BenchUtil.benchCase("message", size) {
          pipe.encryptMessage(plain).fold(e => throw e, _ => ())
        }
    }

File: bindings/kotlin/bench/BenchMessage.kt
// encryptMessage throughput vs plaintext size (Single Message
// profile) at 1 MiB / 16 MiB / 64 MiB.

package io.github.everanium.itb3.kotlin.bench

import io.github.everanium.itb3.kotlin.Pipeline

internal object BenchMessage {

    fun run() {
        Pipeline.init(
            BenchUtil.profileName("singlemsg-triple-nomac-v1"), BenchUtil.buildOpts(),
        ).use { pipe ->
            BenchUtil.header()
            for (size in BenchUtil.sizes) {
                val plain = ByteArray(size)
                BenchUtil.csprngFill(plain)
                BenchUtil.case("message", size) { pipe.encryptMessage(plain) }
                // Pre-encrypt one wire outside the decrypt timing loop.
                val decWire = pipe.encryptMessage(plain)
                BenchUtil.case("message-dec", size) { pipe.decryptMessage(decWire) }
            }
        }
    }
}
